app/db-exists/main.cpp

```diff
@@ -0,0 +1,158 @@
+#include <QtCore/QCoreApplication>
```

Comment:
Seriously..... this is what inheritance is for.... generalize the code to a single implementation. You have the same code duplicated several times.